### DISCUSSION TITLE
hotkey / keyboard shortcut for removing cloud

### DISCUSSION THREAD

**Discussion by neuralgitwork** [freeplane/freeplane#509] [Help] [answered]
pressing the shortcut for inserting a cloud a second time should just remove the just created cloud no?
it was like that in freemind and that software is a lot older. is there a reason they removed this behaviour?

and why if i decide to make a shortcut for remove cloud myself i cannot bind the remove cloud option? why is it one of the only excluded buttons to make a hotkey for????

**Comment by EdoFro** [comment]
> and why if i decide to make a shortcut for remove cloud myself i cannot bind the remove cloud option? why is it one of the only excluded buttons to make a hotkey for????

You can. You should select a node with a cloud first and then you are able to bind a shortcut to the remove cloud command

hope it helps

**Reply by neuralgitwork** [reply]
ok i get it thank you!

using the same shortcut for making cloud, and then with same shortcut removing it is impossible, right?

**Reply by macmarrum** [reply]
It's possible. You just need to use a script, like this one: [toggleCloud.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/toggleCloud.groovy)

f you're new to the topic of scripts in Freeplane, please see https://docs.freeplane.org/scripting/Scripting.html

The script will appear in a new manu "Mac2". If you'd rather have it in Tools->Scripts, remove ="/menu_bar/Mac2" from the script (in the first line).

**Reply by neuralgitwork** [reply]
Thank you so so much for your detailed answer!
This helps a ton !!!

**Comment by dpolivaev** [comment, marked as answer]
The decision to introduce a separate Remove Cloud action was made because instead of a single action for the cloud there are different actions where you can select the form of the cloud and a single remove cloud action. Regarder assigning the shortcut Edo gave a correct explanation above.

**Reply by neuralgitwork** [reply]
thank you!!
